src/components/Form.js

```diff
@@ -104,6 +104,10 @@ const reducer = (draft, action) => {
         config: draft.config,
       };
     }
+    case 'RESET': {
+      window.location.reload();
+      break;
+    }
```

Comment:
Sorry, I didn't make myself clear enough.

There should *not* be a `window.location.reload()` anywhere, ideally, but especially not in a reducer because a reducer is responsible for producing the next state based on the dispatched action and the previous state.

A reset here should instead do something along the lines of `return initialState`, so that the "next" state is effectively a clean/empty state as if the component was just mounted.

The component calls the `useImmerReducer` hook here:

```js
  // load the state management/reducer
  const initialStateFromProps = {...initialState, config, step: steps[0]};
  const [state, dispatch] = useImmerReducer(reducer, initialStateFromProps);
```

So, to reset the state, we basically want to do something along the lines of: `dispatch({type: 'reset', payload: initialStateFromProps})`

and the reducer bit should then look something along the lines of:

```suggestion
    case 'RESET': {
      const initialStateFromProps = action.payload;
      return initialStateFromProps; 
    }
```

The reducer works in a couple of possible modes:

1. You return something. That return value is used as the next state (and immer makes it immutable).
2. You mutate the draft and do or don't `break`. If you don't `break`, the other `case`s below will also evaluate. Then immer takes the mutated draft and essentially makes that the next state.

So, in most cases a reset action is essentially `return initialState`, here it's a bit more complex because the actual initial state has a dynamic aspect from passed in props, but we can capture all that in the callback closures.